pkg/apis/kubernetes/v1/crud_test.go

```diff
@@ -0,0 +1,334 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	"go.anx.io/go-anxcloud/pkg/apis/common/gs"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+	"go.anx.io/go-anxcloud/pkg/client"
+	"go.anx.io/go-anxcloud/pkg/utils/pointer"
+	testutils "go.anx.io/go-anxcloud/pkg/utils/test"
+)
+
+const (
+	mockKubernetesVersion = "1.23.6"
+	nonExistingIdentifier = "non-existing-identifier"
+)
+
+var mockStateOK = map[string]interface{}{"type": gs.StateTypeOK}
+
+var _ = Describe("CRUD", Ordered, func() {
+	var (
+		a   api.API
+		srv *ghttp.Server
+
+		clusterIdentifier  string
+		nodePoolIdentifier string
+
+		clusterName  = "go-anxcloud-" + testutils.RandomHostname()
+		nodePoolName = "go-anxcloud-" + testutils.RandomHostname()
+		location     = corev1.Location{Identifier: "52b5f6b2fd3a4a7eaaedf1a7c019e9ea"} // ANX04
+	)
+
+	BeforeEach(func() {
+		srv = ghttp.NewServer()
```

Comment:
this is only needed for non-integration tests, correct?

Reply:
Correct. Moved to be only initialized on mock tests